### DISCUSSION TITLE
Can I use mix of both client-side and server-side flow with Ory Kratos?

### DISCUSSION THREAD

**Discussion by benjivu** [ory/kratos#1760] [Ideas]
My use case is I have a SPA that uses ReactJS. I know that Ory Kratos supports client-side browser flow but in my use case, I also want to integrate with Hydra so it requires me to have a server-side UI component.

My solution is creating a server-side UI for login, logout, and consent endpoints and the rest of the self-service flows (registration) will be put on the client-side (my ReactJS app).

But I am not sure if this solution is possible or introducing new challenges in the feature? Hope to have some suggestions from you guys in my use case.

**Comment by vinckr** [comment]
Keep in mind that 
`The Login HTML Form cannot be only a Single Page App (Client-side browser application) or a Mobile App! There has to be a server-side component with access to ORY Hydra's Admin Endpoint!`
&
> The admin endpoint allows you to set the user of the token. If it is public (a SPA is an untrusted app that runs in the browser because the user can modify the javascript, a server app runs in a trusted context because the code is running on the server) then anyone could say „I am vincent!“. So it is a terrible idea to run it with a SPA.
> You could write a server app/backend which does the login and sessions etc. and a SPA that interacts with the backend.
> But the SPA should never, ever, be able to talk to Hydra admin directly!

Otherwise, I see no obvious challenges right now, I think it sounds like a great project and definitely keep us in the loop on your progress.

PS: I have moved this to `Ideas` it seemed more fitting.
